transcom/mymove
# INT B-18911 gun safe backend - Prime info addition

## [B-18910](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18910)

## Other INT PR
#12440 

## Summary

This PR adds onto the other INT PR to make the gunSafe entitlement visible to the prime on the backend.

It also adjusts the database column for "gun_safe" to be not nullable, it now must be either true or (by default) false.

### How to test

1. Find a move ID for any *available to Prime* move in the database using DBeaver, and edit it's gun safe flag to "true". Alternatively, if you want to use postman, using the move ID, issue a PATCH request with "gunSafe: true" in the body to "{{baseUrl}}/move-task-orders/:moveTaskOrderID/post-counseling-info".

2. Using Postman, access the Prime API and hit the {{baseUrl}}/move-task-orders/:moveID GET endpoint after getting the move id from the Network devtools tab or from a database explorer, inside the "entitlements" object, you should see a "gunSafe" boolean.

You can also verify via Postman that any API endpoint that uses an "Order" object now has a "gunSafe" boolean inside its nested "entitlements" object.

*Note*: You'll only see the gunSafe field in the object if gunSafe is set to true, otherwise it is not present.
